File: template/CodeSmith/CodeGenerator/02-Result/Presentation/Typings/app/systems/resource/model/resource-query.ts
import { TreeQueryParameter } from '../../../../util';

/**
 * 资源查询参数
 */
export class ResourceQuery extends TreeQueryParameter {
    /**
     * 资源标识
     */
    resourceId;
    /**
     * 应用程序标识
     */
    applicationId;
    /**
     * 资源标识
     */
    uri;
    /**
     * 资源名称
     */
    name;
    /**
     * 资源类型
     */
    type;
    /**
     * 备注
     */
    remark;
    /**
     * 拼音简码
     */
    pinYin;
    /**
     * 扩展
     */
    extend;
    /**
     * 起始创建时间
     */
    beginCreationTime;
    /**
     * 结束创建时间
     */
    endCreationTime;
    /**
     * 创建人编号
     */
    creatorId;
    /**
     * 起始最后修改时间
     */
    beginLastModificationTime;
    /**
     * 结束最后修改时间
     */
    endLastModificationTime;
    /**
     * 最后修改人编号
     */
    lastModifierId;
}

File: src/KissU.Modules/KissU.Modules.GreatWall/KissU.Modules.GreatWall.Application/Queries/ResourceQuery.cs
using System;
using System.ComponentModel.DataAnnotations;
using KissU.Util.Datas.Queries.Trees;

namespace KissU.Modules.GreatWall.Application.Queries
{
    /// <summary>
    /// 资源查询参数
    /// </summary>
    public class ResourceQuery : TreeQueryParameter
    {
        private string _name = string.Empty;

        private string _remark = string.Empty;

        private string _uri = string.Empty;

        /// <summary>
        /// 标识
        /// </summary>
        public Guid? ResourceId { get; set; }

        /// <summary>
        /// 应用程序标识
        /// </summary>
        public Guid? ApplicationId { get; set; }

        /// <summary>
        /// 角色标识
        /// </summary>
        public Guid? RoleId { get; set; }

        /// <summary>
        /// 资源标识
        /// </summary>
        [Display(Name = "资源标识")]
        public string Uri
        {
            get => _uri == null ? string.Empty : _uri.Trim();
            set => _uri = value;
        }

        /// <summary>
        /// 资源名称
        /// </summary>
        [Display(Name = "资源名称")]
        public string Name
        {
            get => _name == null ? string.Empty : _name.Trim();
            set => _name = value;
        }

        /// <summary>
        /// 备注
        /// </summary>
        [Display(Name = "备注")]
        public string Remark
        {
            get => _remark == null ? string.Empty : _remark.Trim();
            set => _remark = value;
        }

        /// <summary>
        /// 起始创建时间
        /// </summary>
        [Display(Name = "起始创建时间")]
        public DateTime? BeginCreationTime { get; set; }

        /// <summary>
        /// 结束创建时间
        /// </summary>
        [Display(Name = "结束创建时间")]
        public DateTime? EndCreationTime { get; set; }
    }
}